src/Avalonia.Controls/NumericUpDown/NumericUpDown.cs

```diff
@@ -678,7 +680,9 @@ private void OnDecrement()
             }
             else
             {
-                result = Maximum;
+                // if Maximum is set we set value to Maximum on decrement. 
+                // otherwise we set value to 0. It ill be clamped to be between Minimum and Maximum later, so we don't need to do it here. 
+                result = IsSet(MaximumProperty) ? Maximum : 0;
```

Comment:
If maximum is not set, but minimum is set to 1, then it won't produce correct value.

Reply:
I think it does as the value is clamped the line below. I wanted to avoid having that calculated twice. I also added such a case to the tests. Or is my test wrong?

Reply:
Oh, right. Haven't paid enough attention